lua/parrot/tests/config_utils_spec.lua

```diff
@@ -38,4 +44,69 @@ describe("config_utils", function()
       }, result)
     end)
   end)
+
+  describe('merge_agent_type', function()
+    it('should merge default and user agents', function()
+      local default_agents = {
+        {
+          name = "ChatGPT4",
+          model = { model = "gpt-4-turbo", temperature = 1.1, top_p = 1 },
+          system_prompt = "You are a versatile AI assistant",
+          provider = "openai",
+        },
+        {
+          name = "ChatGPT3.5",
+          model = { model = "gpt-3.5-turbo", temperature = 1.1, top_p = 1 },
+          system_prompt = "You are a versatile AI assistant",
+          provider = "openai",
+        },
+        {
+          name = "Codestral",
+          model = { model = "codestral-latest", temperature = 1.5, top_p = 1 },
+          system_prompt = "You are a versatile AI assistant",
+          provider = "mistral",
+        },
+        {
+          name = "Mistral-Tiny",
+          model = { model = "mistral-tiny", temperature = 1.5, top_p = 1 },
+          system_prompt = "You are a versatile AI assistant",
+          provider = "mistral",
+        },
+      }
+      local user_agents = {
+        {
+          name = "CustomChatGPT4",
+          model = { model = "gpt-4-turbo", temperature = 1.2 },
+          system_prompt = "You are an AI assistant",
+          provider = "openai",
+        },
+      }
+      local user_providers = {
+        openai = { api_key = '123' },
+      }
+
+      local result = config_utils.merge_agent_type(default_agents, user_agents, user_providers)
+
+      assert.are.same({
+        {
+          name = "CustomChatGPT4",
+          model = { model = "gpt-4-turbo", temperature = 1.2 },
```

Comment:
@frankroeder at this point our tests become helpful in discussing what behaviour we are expecting.

Is `model = { model = "gpt-4-turbo", temperature = 1.2 }` correct? (because it is what the user has configured)

Or should it have been `model = { model = "gpt-4-turbo", temperature = 1.2, top_p = 1 }` (the default should be merged in).

Also the first option could be considered 'invalid' (`top_p` is missing).

Reply:
All that is required is the model. This can simply be defined as a string with `model = "gpt-4-turbo"`. Multiple instances within the code make similar distinctions. This is essentially an inheritance from the fork gp.nvim.

```lua
   local user_agents = {
        {
          name = "CustomChatGPT4",
          model = "gpt-4-turbo",
          system_prompt = "You are an AI assistant",
          provider = "openai",
        },
      }
```

Reply:
> All that is required is the model. This can simply be defined as a string with `model = "gpt-4-turbo"`.

Thanks, in that case this test describes the expected behaviour.

Reply:
I discovered that `M.agents` as a data structure a bit different than the default agents list.
The config_utils function `index_agents_by_name` fixes that.

@frankroeder at this point it at least works for me. I still have to investigate what needs to be cleaned up.